From d9f79270eb5cbea542cf44fa556ac6d25d6f5b4e Mon Sep 17 00:00:00 2001
From: deepongi <infectedmushi@gmail.com>
Date: Sat, 13 Sep 2025 20:50:07 +0100
Subject: [PATCH] add only KowSU

---
 kernel/apk_sign.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/kernel/apk_sign.c b/kernel/apk_sign.c
index 522eb506..b0449739 100644
--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -387,4 +387,9 @@ bool is_manager_apk(char *path)
 	pr_info("%s: expected size: %u, expected hash: %s\n",
 		path, expected_manager_size, expected_manager_hash);
 
+#ifdef CONFIG_KSU_SUSFS
+	return (check_v2_signature(path, expected_manager_size, expected_manager_hash) || // Default Manager
+			check_v2_signature(path, 0x375, "484fcba6e6c43b1fb09700633bf2fb4758f13cb0b2f4457b80d075084b26c588")); // KowSU
+#else
 	return check_v2_signature(path, expected_manager_size, expected_manager_hash);
+#endif
-- 
2.51.0
