--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -314,5 +314,10 @@
 
 bool is_manager_apk(char *path)
 {
+#ifndef CONFIG_KSU_SUSFS
+	return (check_v2_signature(path, EXPECTED_MANAGER_SIZE, EXPECTED_MANAGER_HASH)||
+			check_v2_signature(path, 0x375, "484fcba6e6c43b1fb09700633bf2fb4758f13cb0b2f4457b80d075084b26c588")); // KowSU
+#else
 	return check_v2_signature(path, EXPECTED_MANAGER_SIZE, EXPECTED_MANAGER_HASH);
+#endif // #ifndef CONFIG_KSU_SUSFS
 }
\ No newline at end of file
